index.js

```diff
@@ -220,11 +239,12 @@ function isPathEqual(path1, path2) {
 }
 
 // build a new edge object
-function getNewEdge(fromNode, toNode, weight) {
+function getNewEdge(fromNode, toNode, weight, edgeObject) {
     return {
         fromNode: fromNode,
         toNode: toNode,
-        weight: weight
+        weight: weight,
+        edgeObj: edgeObject
```

Comment:
About the edgeObj in the output, I wonder if it really needed.
to my understanding, the edge object can be extracted by the user (if needed) from the original graph, no need to embed the objects in the library output, right?
also It might ensure backwards compatibility for anyone who parse the current output format

Reply:
So that we would only then store object in the weight parameter if user so decides? That would be completely fine from my point! I probably only added the object to make sure that weight was always integer as in your previous implementation but having the custom weight function would anyways make it possible to control by the user.

So from my side you can remove the edgeObj and use weight as an object

Reply:
I don't think we need to return weight as an object as well,
let me know if I'm wrong, but the output as is right now:
``` javascript
    [{ 
        "totalCost": 5,
        "edges": [
            { "fromNode": "C", "toNode": "E", "weight": 2 },
            { "fromNode": "E", "toNode": "F", "weight": 2 },
            { "fromNode": "F", "toNode": "H", "weight": 1}
        ]
    }]
```
could work even if when you call to ksp with custom weight function.
so even if you have a custom weight function, just pass it to the ksp, and the output will be the same.

so without custom weight function:
```
...
g.setEdge(s, t, 5); // add edge from S to T with weight 5
...
ksp(g, s, t, k); // call to ksp without custom weight fn
// output will be something like: "edges": [ { "fromNode": "s", "toNode": "t", "weight": 5 }] 
````

simallary, with custom weight function:
```
...
g.setEdge(s, t, { foo: "bar", cost: 5 }); // add edge from S to T with custom label obj, weight is now "cost"
...
ksp(g, source, target, k, (e) => g.edge(e).cost); // call to ksp with custom weight fn
// output will be the same as before
```

toughts ?

Reply:
You are right. We will be able to get then full objects directly with standard graphlib functions